packages/client/wallets/aa/src/blockchain/wallets/ZeroDevEip1193Bridge.ts

```diff
@@ -134,13 +135,32 @@ export class ZeroDevEip1193Bridge {
                 const req = ethers.providers.JsonRpcProvider.hexlifyTransaction(params![0]);
                 req.gasLimit = req.gas;
                 delete req.gas;
-                if (req.value === "0x0"){
+                if (req.value === "0x0") {
                     req.value = "0x00";
                 }
                 const tx = await this.signer.sendTransaction(req);
                 return tx.hash;
             }
 
+            case "eth_signTypedData_v4": {
+                if (!this.signer) {
+                    return throwUnsupported("eth_signTypedData_v4 requires a signer");
+                }
+                const signerAddress = params![0];
+                const address = await this.signer.getAddress();
+                if (signerAddress.toLowerCase() !== address.toLowerCase()) {
+                    throw new Error("Signer address does not match provided address");
+                }
+                const typedData = JSON.parse(params![1]);
+                delete typedData.types["EIP712Domain"];
+                const signature = await this.signer._signTypedData(
+                    typedData.domain,
+                    typedData.types,
+                    typedData.message
+                );
```

Comment:
nit, if you want, can help to destructure, but feel free to ignore 
```suggestion
                const { domain, types, message }  = JSON.parse(params![1]);
                delete types["EIP712Domain"];
                const signature = await this.signer._signTypedData(domain, types, message);
```